src/Rules/Comparison/ImpossibleCheckTypeHelper.php

```diff
@@ -147,6 +147,10 @@ public function findSpecifiedType(
 							foreach ($haystackArrayTypes as $haystackArrayType) {
 								if ($haystackArrayType instanceof ConstantArrayType) {
 									foreach ($haystackArrayType->getValueTypes() as $i => $haystackArrayValueType) {
+										if ($haystackArrayValueType instanceof UnionType) {
```

Comment:
This seems weird.

Does it means that 
```
/**
	 * @param array{
	 *     key1: bool|int,
	 * } $myArray
	 */
	public function testBool(array $myArray): ?\stdClass
	{
	    if (\in_array(null, $myArray, true)) {
	        return null;
	    }

		return (object) $myArray;
	}
```
won't be reported too because `bool|int` is an union.